docs/source/conf.py

```diff
@@ -15,24 +15,42 @@
 # https://www.sphinx-doc.org/en/master/usage/configuration.html#general-configuration
 
 extensions = [
+    'sphinx.ext.intersphinx',
     'sphinx_ansible_theme',
-    'sphinxcontrib.apidoc'
+    'sphinx_immaterial',
+    'sphinx_immaterial.apidoc.python.apigen'
 ]
 
-templates_path = ['_templates']
-exclude_patterns = []
-
-#apidoc configuration
-#find more at https://pypi.org/project/sphinxcontrib-apidoc/
-apidoc_module_dir = '../../ansible_sdk/'
-#apidoc_excluded_paths = ['tests']
-apidoc_separate_modules = True
+python_apigen_modules = {
+      "ansible_sdk": "api/",
+      "ansible_sdk.executors": "api/executors/"
+}
 
-source_suffix = {
-    ".rst": "restructuredtext",
-    ".md": "markdown",
+intersphinx_mapping = {
+    'python': ('https://docs.python.org/3', None),
 }
 
+# Temporarily ignore warnings with apidoc generation.
+nitpicky = True
+nitpick_ignore = [
+    ('py:class', 'ansible_sdk.executors.base.AnsibleJobExecutorOptionsBase'),
+    ('py:class', 'ansible_sdk.executors.subprocess._AnsibleContainerJobOptions'),
+    ('py:class', 'ansible_sdk.executors.subprocess._AnsibleContainerJobExecutorBase'),
+    ('py:class', 'ansible_sdk._util.dataclass_compat._DataclassReplaceMixin'),
+    ('py:class', 'ansible_sdk.executors.subprocess._AnsibleContainerJobExecutorBase'),
+    ('py:class', 'ansible_sdk.executors.subprocess._AnsibleContainerJobOptions'),
+    ('py:class', 'ansible_sdk.executors.subprocess._AnsibleContainerJobExecutorBase'),
+    ('py:class', 'ansible_sdk.executors.subprocess._AnsibleContainerJobOptions'),
+    ('py:class', 'ansible_sdk._util.dataclass_compat._DataclassReplaceMixin'),
+    ('py:class', 'ansible_sdk.executors.subprocess._AnsibleContainerJobExecutorBase'),
+    ('py:class', 'ansible_sdk.executors.subprocess.OptionsT'),
+    ('py:class', 'ansible_sdk.executors.subprocess._AnsibleContainerJobOptions'),
+    ('py:class', 'ansible_sdk.executors.mesh.AnsibleMeshJobOptions'),
+]
+
+templates_path = ['_templates']
```

Comment:
This is unnecessary unless you're going to integrate custom template overrides placed under that directory.